diff --git a/Ethernet/wizchip_conf.h b/Ethernet/wizchip_conf.h
index a5d65f7..80916e5 100644
--- a/Ethernet/wizchip_conf.h
+++ b/Ethernet/wizchip_conf.h
@@ -72,7 +72,7 @@ extern "C" {
 #define W5500						5500
 
 #ifndef _WIZCHIP_
-#define _WIZCHIP_                      W5500   // W5100, W5100S, W5200, W5300, W5500
+#define _WIZCHIP_                      W5100S   // W5100, W5100S, W5200, W5300, W5500
 #endif
 
 #define _WIZCHIP_IO_MODE_NONE_         0x0000
